src/model_repository_manager.cc

```diff
@@ -1191,18 +1203,10 @@ ModelRepositoryManager::ModelLifeCycle::Load(
       LOG_INFO << "loading: " << model_name << ":" << version;
       model_info->state_ = ModelReadyState::LOADING;
       model_info->state_reason_.clear();
-      {
-        // FIXME WAR for glibc bug when spawning threads too
-        // quickly. https://sourceware.org/bugzilla/show_bug.cgi?id=19329
-        // We should instead use a thread-pool here with the number of
-        // threads chosen to provide the required amount of load
-        // parallelism. DLIS-1833.
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
-        std::thread worker(
-            &ModelRepositoryManager::ModelLifeCycle::CreateModel, this,
-            model_name, version, model_info);
-        worker.detach();
-      }
+      // Load model asynchronously via thread pool
+      boost::asio::post(*load_pool_, [this, model_name, version, model_info]() {
+        CreateModel(model_name, version, model_info);
+      });
```

Comment:
I was wondering do we need to make sure that all the threads have exited before destructing this class? Especially since the callback is capturing this.

Reply:
@GuanLuo @tanmayv25 what do you think?

Reply:
I think the thread pool destructor will join and stop the working thread, however, we should destruct the thread pool as the first step of stopping model lifecycle as the callback can modify model lifecycle members (updating model status)